src/main/java/net/fabricmc/loader/discovery/DirectoryModCandidateFinder.java

```diff
@@ -51,13 +55,28 @@ public void findCandidates(FabricLoader loader, BiConsumer<URL, Boolean> urlProp
 		}
 
 		try {
-			Files.walk(path, 1, FileVisitOption.FOLLOW_LINKS).forEach((modPath) -> {
-				if (!Files.isDirectory(modPath) && modPath.toString().endsWith(".jar")) {
-					try {
-						urlProposer.accept(UrlUtil.asUrl(modPath), requiresRemap);
-					} catch (UrlConversionException e) {
-						throw new RuntimeException("Failed to convert URL for mod '" + modPath + "'!", e);
+			Files.walkFileTree(this.path, EnumSet.of(FileVisitOption.FOLLOW_LINKS), 1, new SimpleFileVisitor<Path>() {
+				@Override
+				public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+					/*
+					 * We only propose a file as a possible mod in the following scenarios:
+					 * General: Must be a jar file
+					 *
+					 * Some OSes Generate metadata so consider the following because of OSes:
+					 * UNIX: Exclude if file is hidden; this occurs when starting a file name with `.`
+					 * MacOS: Exclude hidden + startsWith "." since Mac OS names their metadata files in the form of `.mod.jar`
+					 */
+
+					if (!Files.isHidden(file) && !file.toString().startsWith(".") && file.toString().endsWith(".jar")) {
```

Comment:
```suggestion
					String fileName = file.toString();

					if (fileName.endsWith(".jar") && !fileName.startsWith(".") && !Files.isHidden(file)) {
```
would be clearer imo.

Reply:
Maybe even file.getFileName().toString() to be safe